qemu: update to 8.1.0

jiegec opened the pull request:
<!-- For description on topic creation and maintenance, please refer to [this Wiki article](https://wiki.aosc.io/developer/packaging/topic-based-maintenance-guideline/). -->

Topic Description
-----------------

<!-- Please input topic description here. -->

- sndio: update to 1.9.0
- virglrenderer: update to 0.10.4
- lzfse: new, 1.0
- libu2f-emu: new, 0+git20200904
- libblkio: new, 1.3.0
- canokey-qemu: new, 0+git20230605
- qemu: update to 8.1.0

Package(s) Affected
-------------------

<!-- Please list all package(s) affected by this topic here. -->

```
sndio virglrenderer lzfse libu2f-emu libblkio canokey-qemu qemu

```

Security Update?
----------------

<!-- If this topic contains security update(s), please uncomment "Yes,"
     mark with the `security` and `priority` labels, and make sure to mark your commits to relevant issue numbers.

     Please see GitHub's documentation on "Linking a pull request to an issue":

     https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->

<!-- Yes - Issue Number: ISSUENUMBER -->
No

<!-- Please uncomment the "Build Order" section if your topic affects more than one package. -->


Build Order
-----------

<!-- Please describe in what order maintainers should build this pull request. You can use the following template, use space to separate packages. -->

```
sndio virglrenderer lzfse libu2f-emu libblkio canokey-qemu qemu
```


Test Build(s) Done
------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->    
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

<!-- Maintainers should review file changes and, if the build script(s) affected complies with the
     [Package Styling Manual](https://wiki.aosc.io/developer/packaging/package-styling-manual/), please
     add `lgtm` label to this issue. -->

<!-- Maintainers and users may now test the packages in this topic and, once user/maintainer feedback indicates that the update(s) work as expected and find its quality satisfactory,
     another maintainer may now review this pull request and mark it as Approved. After which, the maintainer will build affected package(s) and upload them to the `stable` repository. -->

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

<!-- TODO: CI to auto-fill architectural progress. -->

MingcongBai committed:
edk2: drop, orphaned

Just use the ones from Qemu.

MingcongBai committed:
seabios: drop, orphaned

Just use the ones from Qemu.

jiegec committed:
qemu: update to 8.1.0

- Merge in SeaBIOS and EDK2, as Qemu builds these firmware during build-time.
- Survey and specify options explicitly, where applicable.
- Add sndio, ndctl, lzfse, rdma-core, libu2f-emu, canokey-qemu, fuse-3,
  and libblkio deps.
- Drop upstreamed patch 1001.
- Rename option --enable-jemalloc => --enable-malloc=jemalloc.
- Rebase patches 1002, 1004, 1005, and 9999.

Note: --disable-rbd, we do not maintain Ceph.
Note: --disable-glusterfs, we do not maintain GlusterFS.
Note: --disable-selinux, we do not maintain SELinux (yet?).
Note: --disable-libpmem, not ready to maintain that stack yet (libfabric => pmdk).
Note: --disable-cfi, not compatible with many features.
Note: --disable-netmap, not sure how useful it is and it seems difficult to maintain.

Ref: Patch 1001 has been upstreamed at
https://github.com/qemu/qemu/commit/abe45a859b897736d7f428f58d1e5cab4fec4ddf.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

MingcongBai committed:
sndio: update to 1.9.0

This update is needed to enable sndio support for Qemu 8.x.

MingcongBai committed:
virglrenderer: update to 0.10.4

- Specify options explicitly where applicable.
- Use Git source.

Note: -Dplatforms: Platforms support, set to auto to enable all.
Note: Venus and render server are still experimental.

FIXME: minigbm_allocation breaks build at src/vrend_renderer.c
../src/vrend_renderer.c:7887:19: error: implicit declaration of function
‘gbm_bo_get_plane_size’; did you mean ‘gbm_bo_get_plane_count’?
[-Werror=implicit-function-declaration]
7887 |       gr->size += gbm_bo_get_plane_size(bo, plane);
     |                   ^~~~~~~~~~~~~~~~~~~~~
     |                   gbm_bo_get_plane_count

MingcongBai committed:
lzfse: new, 1.0

MingcongBai committed:
libu2f-emu: new, 0+git20200904

MingcongBai committed:
libblkio: new, 1.3.0

MingcongBai committed:
canokey-qemu: new, 0+git20230605

MingcongBai commented:
LGTM on AArch64.

![图片](https://github.com/AOSC-Dev/aosc-os-abbs/assets/5006263/31748ba8-80ef-4276-af13-6c43e9aab1c3)

jiegec commented:
rdma-core does not build on riscv64:

```
[INFO]:  Running post-build filter: perl ...
[INFO]:  Removed perllocal.pod.
[INFO]:  Removed .packlist.
[INFO]:  Running post-build filter: retro_drop_docs ...
[DEBUG]: Member 'mainline' in Architecture group 'mainline' mismatches retro.
[INFO]:  All architecture group members 'mainline' mismatches retro: taking false branch.
[INFO]:  Non-Retro architectures detected, skipping retro_drop_docs ...
[INFO]:  Found autobuild/conffiles - good job! Content as follows ...
/etc/infiniband-diags/error_thresholds
/etc/infiniband-diags/ibdiag.conf
/etc/iwpmd.conf
/etc/modprobe.d/mlx4.conf
/etc/modprobe.d/truescale.conf
/etc/rdma/mlx4.conf
/etc/rdma/modules/infiniband.conf
/etc/rdma/modules/iwarp.conf
/etc/rdma/modules/iwpmd.conf
/etc/rdma/modules/opa.conf
/etc/rdma/modules/rdma.conf
/etc/rdma/modules/roce.conf
/etc/rdma/modules/srp_daemon.conf
/etc/rdma/rdma.conf
/etc/srp_daemon.conf
[WARN]:  However, there seems to be more/less files found in $PKGDIR/etc ...
[WARN]:  Please compare the following list of files against the current autobuild/conffiles ...
/etc/infiniband-diags/error_thresholds
/etc/infiniband-diags/ibdiag.conf
/etc/iwpmd.conf
/etc/libibverbs.d/hfi1verbs.driver
/etc/libibverbs.d/ipathverbs.driver
/etc/libibverbs.d/rxe.driver
/etc/libibverbs.d/siw.driver
/etc/modprobe.d/truescale.conf
/etc/rdma/mlx4.conf
/etc/rdma/modules/infiniband.conf
/etc/rdma/modules/iwarp.conf
/etc/rdma/modules/iwpmd.conf
/etc/rdma/modules/opa.conf
/etc/rdma/modules/rdma.conf
/etc/rdma/modules/roce.conf
/etc/rdma/modules/srp_daemon.conf
/etc/rdma/rdma.conf
/etc/srp_daemon.conf
[INFO]:  Creating empty postinst.
[INFO]:  Creating empty prerm.
[INFO]:  Creating empty postrm.
[INFO]:  Creating empty preinst.
[INFO]:  No alternatives file found.
[INFO]:  Running post-build QA tests ...
[WARN]:  QA (W322): Zero-byte files found:

/var/cache/acbs/build/acbs.bi1ew0zu/rdma-core-38.0/abdist/usr/lib/python3.10/pyverbs/__init__.py

[WARN]:  QA warning(s) found, please refer to abqawarn.log ...
[INFO]:  Building dpkg package(s) ...
dpkg-deb: error: conffile '/etc/modprobe.d/mlx4.conf' does not appear in package
[ERROR]: Build error detected ^o^

```
